src/software/ai/navigator/path_planner/theta_star_path_planner.h

```diff
@@ -22,111 +22,174 @@ class ThetaStarPathPlanner : public PathPlanner
      * @param obstacles obstacles to avoid
      *
      * @return a vector of points that is the optimal path avoiding obstacles
-     * 		if no valid path then return empty vector
+     *         if no valid path then return empty vector
      */
     std::optional<Path> findPath(const Point &start, const Point &destination,
                                  const Rectangle &navigable_area,
                                  const std::vector<Obstacle> &obstacles) override;
 
    private:
-    typedef std::pair<int, int> CellCoordinate;
-    typedef std::pair<double, CellCoordinate> OpenListCell;
+    class Coordinate : public std::pair<int, int>
+    {
+       public:
+        Coordinate(int row, int col) : std::pair<int, int>(row, col) {}
+
+        Coordinate() : std::pair<int, int>() {}
+
+        int row(void)
+        {
+            return this->first;
+        }
+
+        int col(void)
+        {
+            return this->second;
+        }
+    };
 
-    class GridCell
+    class CellHeuristic
     {
        public:
-        GridCell(int parent_i, int parent_j, double f, double g, double h)
-            : parent_i_(parent_i), parent_j_(parent_j), f_(f), g_(g), h_(h)
+        CellHeuristic() {}
+
+        /**
+         * Updates and initializes CellHeuristics internal variables
+         *
+         * @param parent parent
+         * @param f f value
+         * @param g g value
+         */
+        void updateAndInitialize(Coordinate parent, double f, double g)
```

Comment:
nit; `initialize` generally refers to the first time that you set a value, I would rename this to `update`.